### DISCUSSION TITLE
Adding SMB file share to servers

### DISCUSSION THREAD

**Discussion by tomyki** [datamesh-architecture/datacontract-specification#103] [Ideas]
Hello,
Is it possible to add support for SMB file share option in servers? Now I cannot find anything resembling that

**Comment by jochenchrist** [comment]
Can you prepare an example how this server block would look like.

**Reply by tomyki** [reply]
I'd propose something like that:

> "SMBServer": {
    "type": "object",
    "title": "SMBServer",
    "properties": {
      "account": {
        "type": "string",
        "description": "An optional string describing the account."
      },
      "server": {
        "type": "string",
        "description": "A required string describing the server."
      },
      "share": {
        "type": "string",
        "description": "A required string describing the share."
      },
      "location": {
        "type": "string",
        "description": "A required string describing the location of file."
      },
      "port": {
        "type": "integer",
        "description": "An optional string describing the smb server port.",
        "examples": [
          445
        ]
      },
    "additionalProperties": true,
    "required": [
      "server",
      "share",
      "location"
    ]
  },


Thanks in advance!

**Comment by MrMikeFloyd** [comment]
I'd appreciate this being implemented as we have the same requirement :)
